ibis/backends/pyspark/compiler.py

```diff
@@ -157,7 +158,17 @@ def compile_selection(t, expr, scope, timecontext, **kwargs):
 
     for predicate in op.predicates:
         col = t.translate(predicate, scope, timecontext)
-        result_table = result_table[col]
+        # Due to an upstream Spark issue (SPARK-33057) we cannot
+        # directly use filter with a window operation. The workaround
+        # here is to assign a temporary column for the filter predicate,
+        # do the filtering, and then drop the temporary column.
+        try:
+            filter_column = 'internal_column_for_filtering_only'
+            result_table = result_table.withColumn(filter_column, col)
+            result_table = result_table.filter(F.col(filter_column))
+            result_table = result_table.drop(F.col(filter_column))
+        except AnalysisException:
```

Comment:
What is this?

Reply:
Removed and replaced with change in `col_in_selection_order`.